spec/VectorBasemapLayerSpec.js

```diff
@@ -43,13 +46,144 @@ describe('VectorBasemapLayer', function () {
+    const layer = new L.esri.Vector.vectorBasemapLayer(basemapKeyV2, {
+      apikey: apikey,
+      version:2
+    });
+
+    expect(layer.options.version).to.equal(2);
+  });
+
+  it('should set the service version to \'1\' when a version is not provided', function () {
+    const layer = new L.esri.Vector.vectorBasemapLayer(basemapKey, {
+      apikey: apikey
+    })
+
+    expect(layer.options.version).to.equal(1);
+  });
+
+  it('should save the language from the constructor', function () {
+    const layer = new L.esri.Vector.vectorBasemapLayer(basemapKeyV2, {
+      apikey: apikey,
+      version:2,
+      language:language
+    })
+
+    expect(layer.options.language).to.equal(language);
+  });
+
+  it('should error if a language is provided when accessing the v1 service', function () {
+    expect(function () {
+      L.esri.Vector.vectorBasemapLayer(basemapKey, {
+        apikey:apikey,
+        language:language
+      });
+    }).to.throw('The language parameter is only supported by the basemap styles service v2. Provide a v2 style enumeration to use this option.');
+  });
+
+  it('should not accept a v2 style enumeration when accessing the v1 service', function () {
+    expect(function () {
+      L.esri.Vector.vectorBasemapLayer(basemapKeyV2, {
+        apikey:apikey,
+        version:1
+      });
+    }).to.throw(basemapKeyV2 + ' is a v2 style enumeration. Set version:2 to request this style')
+  })
+
+  it('should not accept a v1 style enumeration when accessing the v2 service', function () {
+    expect(function () {
+      L.esri.Vector.vectorBasemapLayer(basemapKey, {
+        apikey:apikey,
+        version:2
+      });
+    }).to.throw(basemapKey + ' is a v1 style enumeration. Set version:1 to request this style')
+  })
+
+  it('should load a custom basemap style from an item ID when using the v1 service', function () {
+    const customLayer =  L.esri.Vector.vectorBasemapLayer(customBasemap, {
+      apikey:apikey,
+      version:1
+    })
+    expect(customLayer._maplibreGL.options.style).to.equal(`https://basemaps-api.arcgis.com/arcgis/rest/services/styles/${customBasemap}?type=style&token=${apikey}`)
+  })
+
+  it('should load a custom basemap style from an item ID when using the v2 service', function () {
+    const customLayer =  L.esri.Vector.vectorBasemapLayer(customBasemap, {
+      apikey:apikey,
+      version:2
+    })
+    expect(customLayer._maplibreGL.options.style).to.equal(`https://basemapstyles-api.arcgis.com/arcgis/rest/services/styles/v2/styles/items/${customBasemap}?token=${apikey}`)
+  })
+
+  it('should error if a language is provided when loading a custom basemap style', function () {
+    expect(function () {
+      L.esri.Vector.vectorBasemapLayer(customBasemap, {
+        apikey,apikey,
+        version:2,
+        language:language
+      })
+    }).to.throw('The \'language\' parameter is not supported for custom basemap styles')
+  })
+
```

Comment:
I don't see a unit test for the new functionality "it should load a v2 basemap if using a v2 basemap key enum even if not passing the `version: 2` property" - can you add a unit test for that case?